### DISCUSSION TITLE
Inquiry about the extremely small value in bar chart

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#633] [Questions] [answered]
Hi, I wonder if we have a way to render extremely small value in the bar chart, showing it in a pill shape, e.g. fully rounded for every corner. In my example below, it is showing like with a flat top and bottom. Is there any way to tweak that style? 
<img width="191" alt="Screenshot 2024-03-26 at 9 19 03 PM" src="https://github.com/patrykandpatrick/vico/assets/160801134/65a4a0dd-33f6-4d55-822c-ba12d395e2ec">

**Comment by patrickmichalik** [comment]
Hello! Could you elaborate on the desired result? The corners of the blue column on the left appear to be fully rounded. The top and bottom are flat because the width of the column is greater than its height. The column would have to be taller for this not to occur. Are you looking to set a minimum column height, to achieve the result shown in the image below, or to do something else?

![](https://github.com/patrykandpatrick/vico/assets/120058021/593e76b5-b938-4810-99a7-ec24e1683653)

**Reply by siyingch86** [reply]
Hi @patrickmichalik is there any way we could set a minimum column height?

**Reply by patrickmichalik** [reply]
Please see [here](https://github.com/patrykandpatrick/vico/discussions/633#discussioncomment-9030301), @siyingch86.

**Comment by siyingch86** [comment]
Yes, I would want to make it as the picture you showed. Maybe could be achieved by setting a minimum column height while not changing the real value behind. Let's say we have a bar of width N, and we want any time the value reaches below N/2, we don't automatically render it into a flat shape. Instead, we show it as a fixed height of N/2. In this case, the y value not changed. User can have a better visual experience and if they want to access the exact value, they can click on the bar and view them through the marker. Is it possible to set this in Vico?

**Reply by patrickmichalik** [reply, marked as answer]
Got it. You can implement such behavior by creating a `LineComponent` subclass. `drawVertical` will have to be overridden.
